<launch>
  
  <arg name="record_videoray" default="false" />
  <arg name="gui_enable" default="true" />

  <group ns="videoray">
  
    <!-- Primary VideoRay Control Node -->
    <remap from="pose" to="/rqt_compass/pose" />
    <remap from="pose_only" to="/pose_only" />
    <remap from="throttle_cmd" to="/rqt_thrust_monitor/throttle_cmd" />
    <remap from="desired_trajectory" to="/rqt_autopilot/desired_trajectory" />
    <remap from="sonar_enable_log" to="rqt_blueview/sonar_enable_log" />
    <remap from="videoray_status" to="/rqt_rov_status/videoray_status" />
    <remap from="uhri_comm" to="/rqt_uhri/uhri_comm" />
    <remap from="experiment_notes" to="/rqt_experiment_notes/experiment_notes" />
    <node pkg="videoray" name="videoray_control" type="control" output="screen">
      <param name="tick_rate" type="double" value="50" />
      <param name="save_directory" type="string" value="/home/syllogismrxs/sonar_log" />
    </node>
   
    <!-- Joystick Control Node -->
    <param name="joystick_dev" value="/dev/input/js2" />
    <node pkg="input" name="joystick_1" type="joystick" output="screen">
      <param name="tick_rate" type="double" value="100" />
    </node>
  
    <!-- Include Analog Video Camera -->
    <include file="$(find videoray)/launch/camera.launch" />

  </group>

  <!-- rosbag - log data node -->
  <!-- <node pkg="rosbag" type="record" name="rosbag_record_videoray"
        args="-a -o /tmp/videoray-ros-data"
        if="$(arg record_videoray)" /> -->

  <node pkg="rosbag" type="record" name="rosbag_record_videoray"
        args="-b 0 /pose_only /rosout /rosout_agg /rqt_autopilot/desired_trajectory /rqt_compass/pose /rqt_thrust_monitor/throttle_cmd /videoray/accelerations /videoray/camera/image_raw /videoray/joystick /videoray/rqt_blueview/sonar_enable_log /videoray/rqt_blueview/sonar_max_range /videoray/rqt_blueview/sonar_min_range /videoray/rqt_blueview/sonar_thresh /videoray/sonar_image -o /tmp/videoray-ros-data" if="$(arg record_videoray)" />

 
<!--

rosbag record -b 0 -b 0 /pose_ly /rosout /rosout_agg /rqt_autopilot/desired_trajectory /rqt_compass/pose /rqt_thrust_monitor/throttle_cmd /videoray/accelerations /videoray/camera/image_raw /videoray/joystick /videoray/rqt_blueview/sonar_enable_log /videoray/rqt_blueview/sonar_max_range /videoray/rqt_blueview/sonar_min_range /videoray/rqt_blueview/sonar_thresh /videoray/sonar_image
 
/pose_only
/rosout
/rosout_agg
/rqt_autopilot/desired_trajectory
/rqt_compass/pose
/rqt_thrust_monitor/throttle_cmd
/videoray/accelerations
/videoray/camera/image_raw
/videoray/joystick
/videoray/rqt_blueview/sonar_enable_log
/videoray/rqt_blueview/sonar_max_range
/videoray/rqt_blueview/sonar_min_range
/videoray/rqt_blueview/sonar_thresh
/videoray/sonar_image
-->

  <!-- ================ -->
  <!-- Startup the GUIS -->
  <!-- ================ -->
  <group if="$(arg gui_enable)">
    
    <!-- Compass -->
    <node pkg="rqt_compass" name="rqt_compass" type="rqt_compass" />

    <!-- Autopilot -->
    <node pkg="rqt_autopilot" name="rqt_autopilot" type="rqt_autopilot" />

    <!-- ROV Status-->
    <!--<node pkg="rqt_rov_status" name="rqt_rov_status" type="rqt_rov_status" />-->
    
    <!-- UHRI Comm-->
    <node pkg="rqt_uhri" name="rqt_uhri" type="rqt_uhri" />
    
    <!-- Experimental notes-->
    <node pkg="rqt_experiment_notes" name="rqt_experiment_notes" type="rqt_experiment_notes" />
    
    <!-- Thrust monitor -->
    <!-- <node pkg="rqt_thrust_monitor" name="rqt_thrust_monitor" type="rqt_thrust_monitor" /> -->

  </group>  
  
</launch>
